### PR TITLE
GIX-1481: Render last maturity distribution date

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We want to show the last time that maturity was distributed.

This PR implements rendering the date of the last reward in the maturity card.

# Changes

* Render another KeyValuePair in NnsNeuronMaturityCard with the latest maturity distribution date.
* Adapt styles to combine both staked maturity and latest distribution pairs.

# Tests

* Add test in NnsNeuronDetail page to check that last maturity distribution date is rendered.
* Add test in NnsNeuronMaturityCard to check that last maturity distribution date is rendered or not.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UI:

Screenshot 2023-04-14 at 11 16 57

This is when we have also staked maturity:

Screenshot 2023-04-14 at 10 52 23

**Comment by lmuntaner** [comment]
@mstrasinskis @dskloetd please review

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronMaturityCard.svelte [hunk 1]
```diff
@@ -42,13 +42,15 @@
   </KeyValuePair>
 
   {#if hasStakedMaturity(neuron)}
-    <KeyValuePair testId="staked-maturity">
-      <svelte:fragment slot="key">{$i18n.neurons.staked}</svelte:fragment>
+    <div class="details">
```
Comment (lmuntaner) [inline frontend/src/lib/components/sns-neuron-detail/SnsNeuronMaturityCard.svelte:45]: I had to also adapt this file because they both use the same CSS mixin.

And we have the rule of not allowing classes to not be used. Therefore, I had to use the same style here as in NnsNeuronMaturityCard even though it's not yet needed.

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts [hunk 2]
```diff
@@ -20,6 +21,14 @@ import {
 } from "$tests/mocks/auth.store.mock";
 import en from "$tests/mocks/i18n.mock";
 import { mockFullNeuron, mockNeuron } from "$tests/mocks/neurons.mock";
+import { mockRewardEvent } from "$tests/mocks/nns-reward-event.mock";
+// TODO: Fix prettier command. For some reason it moves the JestPageObjectElement import after the KeyValuePairPo import.
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts:25]: Isn't that expected? Lower case `j` (of "jest.page-object") comes after upper case `K` (of "KeyValuePair.page-object") in ascii.

File: frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts [hunk 3]
```diff
@@ -119,4 +126,17 @@ describe("NeuronDetail", () => {
 
     await waitFor(() => expect(querySkeleton(container)).not.toBeNull());
   });
+
+  it("should render last maturity distribution", async () => {
+    const { container } = render(NnsNeuronDetail, props);
+
+    await runResolvedPromises();
+
+    const keyValuePairPo = KeyValuePairPo.under({
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts:135]: Can you get it via the NnsNeuronDetailPo?

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts:25]: Yes, the weird part is that we have the same exact order in the other test `frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts`:

```
import { JestPageObjectElement } from "$tests/page-objects/jest.page-object";
import { KeyValuePairPo } from "$tests/page-objects/KeyValuePair.page-object";
```

And that one is ok.

I added myself a task to research the difference on both files.

File: frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts:135]: sure, I can add it there

Comment (lmuntaner) [reply frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts:135]: I'm never sure how much you want me to add as POs or not add because it's not part of the task at hand.

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronMaturityCard.spec.ts]: No errors now that I added the new POs

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/NnsNeuronDetail.spec.ts]: That makes sense, but it's hard to follow sometimes where the test is already set up and you want to just add a new test.

But in general I think that it's a good rule of thumb to create the PO for X, even though it's used only in one test case.

Does that make sense?
